xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/commentsinline.vm

```diff
@@ -256,10 +250,20 @@ $xwiki.ssfx.use('uicomponents/viewers/comments.css', true)
 ##
 ##
 #macro(displayCommentButton)
-<button type="button" id="openCommentForm" class="btn btn-default">
-  $services.icon.renderHTML('comment')
-  $services.localization.render('core.viewers.comments.add.comment.label')
-</button>
+  <button type="button" id="openCommentForm" class="btn btn-default">
+    $services.icon.renderHTML('comment')
+    #if($xcontext.user != 'XWiki.XWikiGuest')
+      $services.localization.render('core.viewers.comments.add.comment.label')
+    #else
+      $services.localization.render('core.viewers.comments.add.commentAsGuest.label')
+    #end
+  </button>
+  #if($xcontext.user == 'XWiki.XWikiGuest')
+    <a id='loginAndComment' class='btn btn-primary'
+      href='$xwiki.getURL('XWiki.XWikiLogin', 'login', "xredirect=$escapetool.url($doc.getURL('view', '#Comments'))")'>
```

Comment:
Not sure about the redirect URL: I might have used the comment viewer instead (i.e. viewer=comments in the query string)

Reply:
Addressed in https://github.com/xwiki/xwiki-platform/pull/2769/commits/252cf0ee828622814bf1c4349ce584fe49a79d3e :+1: 

Here is the result of the redirect: 
![image](https://github.com/xwiki/xwiki-platform/assets/28761965/457ead78-9e67-46bb-a20e-7851c5043d49)
Looks appropriate :)

Reply:
Note that it will probably fix https://jira.xwiki.org/browse/XWIKI-17513. I linked the ticket in the original JIRA issue you're fixing.